source/dofs/dof_handler_policy.cc

```diff
@@ -1252,82 +1261,104 @@ namespace internal
+                        &quadrants[0],
+                        quadrants_as_chars.size());
 
-            char *ptr = &buffer[0];
+            // now serialize everything
+            ar &quadrants_as_chars
+            &tree_index
+            &dof_numbers_and_indices;
+          }
 
-            const unsigned int num_cells = tree_index.size();
-            std::memcpy(ptr, &num_cells, sizeof(unsigned int));
-            ptr += sizeof(unsigned int);
+          /**
+           * Read the data of this object from a stream for the purpose of
+           * serialization. Throw away the previous content.
+           */
+          template <class Archive>
+          void load (Archive &ar,
+                     const unsigned int /*version*/)
+          {
+            // undo the copying trick from the 'save' function
+            std::vector<char> quadrants_as_chars;
+            ar &quadrants_as_chars
+            &tree_index
+            &dof_numbers_and_indices;
 
-            const unsigned int num_dofs = dof_numbers_and_indices.size();
-            std::memcpy(ptr, &num_dofs, sizeof(unsigned int));
-            ptr += sizeof(unsigned int);
+            quadrants.resize (quadrants_as_chars.size() / sizeof(quadrants[0]));
+            std::memcpy(&quadrants[0],
+                        &quadrants_as_chars[0],
+                        quadrants_as_chars.size());
+          }
 
-            std::memcpy(ptr,
-                        &tree_index[0],
-                        num_cells*sizeof(unsigned int));
-            ptr += num_cells*sizeof(unsigned int);
+          BOOST_SERIALIZATION_SPLIT_MEMBER()
 
-            std::memcpy(ptr,
-                        &quadrants[0],
-                        num_cells * sizeof(typename dealii::internal::p4est::
-                                           types<dim>::quadrant));
-            ptr += num_cells*sizeof(typename dealii::internal::p4est::types<dim>::
-                                    quadrant);
 
-            if (num_dofs>0)
-              std::memcpy(ptr,
-                          &dof_numbers_and_indices[0],
-                          dof_numbers_and_indices.size() * sizeof(dealii::types::global_dof_index));
-            ptr += dof_numbers_and_indices.size() * sizeof(dealii::types::global_dof_index);
+          /**
+           * Pack the data that corresponds to this object into a buffer in
+           * the form of a vector of chars and return it.
+           */
+          std::vector<char>
+          pack_data () const
+          {
+            // set up a buffer and then use it as the target of a compressing
+            // stream into which we serialize the current object
+            std::vector<char> buffer;
+            {
+              boost::iostreams::filtering_ostream out;
+              out.push(boost::iostreams::gzip_compressor
+                       (boost::iostreams::gzip_params
+                        (boost::iostreams::gzip::best_compression)));
```

Comment:
Yes, I agree it doesn't matter. My primary motivation was to get rid of the hand-serialization and use the BOOST facilities for that. That's so much more reliable.

But then I thought that while I'm there, I might as well save ourselves the few bytes. It is not going to matter because (i) the amount of data we send and receive here in the ghost cell exchange is small, and (ii) the latency of any communication is likely orders of magnitude larger than the time it takes us here to compress the few thousand bytes. (I'm pretty sure that what matters here for small amounts of data is not the bandwidth, but the latency.)